compiler/src/dotty/tools/repl/ReplDriver.scala

```diff
@@ -51,9 +51,15 @@ case class State(objectIndex: Int,
                  imports: Map[Int, List[tpd.Import]],
                  context: Context)
 
-/** Main REPL instance, orchestrating input, compilation and presentation */
+/** Main REPL instance, orchestrating input, compilation and presentation.
+ *
+ * @param out where regular output (eg. "define trait Foo") is printed
+ * @param errorReporter if set, this reporter is passed to the parser and typer. Otherwise,
+ *        the warnings and errors are printed to `out` as well.
+ */
 class ReplDriver(settings: Array[String],
                  out: PrintStream = Console.out,
+                 errorReporter: Option[() => Reporter] = None,
```

Comment:
Is there a good reason why this should be a function and not a `Reporter`?

Reply:
This is too long ago to actually remember, but my guess would be that we might need a new instance with each run?